src/utilities.jl

```diff
@@ -0,0 +1,45 @@
+"""
+    format_message_counts(i, n)
+
+If `n==1` returns `""`, otherwise `"\n\n[\$i / \$n]"`.
+"""
+function format_message_counts(i, n)
+    return n == 1 ? "" : "\n\n[$i / $n]"
+end
+
+# Combine `N` strings into `M <= N` strings, by concatenating consecutive strings, such that
+# each of the resulting strings has length `<= max_length`, unless the input string was already
+# longer than `max_length` , in which case it is passed on.
+#
+# The purpose of this is for nicely splitting long lists of attachments.
+function combine_texts(texts; max_length=3800, format_message_counts=format_message_counts)
```

Comment:
As `format_message_counts` is being passed in you may want to pass in the message to allow for further customization or just narrow the scope of this keyword to be something less generic

Reply:
Should `max_length` be a constant? I assume it's based upon a restriction in Slack

Reply:
yeah, it's to be under 4k characters, which is a Slack limitation. I chose 3800 so that `format_message_counts` likely doesn't exceed it, and because I'm not sure exactly what Slack means by a character (a grapheme? a codeunit? etc), so might as well add some wiggle room. I made it a keyword argument so I could test it easier.

> As format_message_counts is being passed in you may want to pass in the message to allow for further customization or just narrow the scope of this keyword to be something less generic

I don't quite get what you mean? Do you mean we should rename it here as well to `message_count_suffix`?

Reply:
Ah ok, I've renamed it to `message_count_suffix` now. I don't see a need for something more general right now but we could do that later if the need arises. Do you have a need for something more general?

Reply:
I don't have specific need for something more generic. Changing the keyword to match the function scope works for me!